addons/mail/models/discuss/discuss_channel_member.py

```diff
@@ -85,52 +120,66 @@ def create(self, vals_list):
         partner will be added in the channel and the security rule will always authorize
         the creation.
         """
-        if not self.env.is_admin() and not self.env.context.get('mail_create_bypass_create_check') is self._bypass_create_check:
-            for vals in vals_list:
-                if 'channel_id' in vals:
-                    channel_id = self.env['discuss.channel'].browse(vals['channel_id'])
-                    if not channel_id._can_invite(vals.get('partner_id')):
-                        raise AccessError(_('This user can not be added in this channel'))
+        if self.env.context.get("mail_create_bypass_create_check") is self._bypass_create_check:
+            self = self.sudo()
```

Comment:
Note: `_bypass_create_check` from channel creation should bypass ACL that prevent from adding the current user in any channel, but it should not bypasss channel type specific checks like it did before.